src/urllib3/request.py

```diff
@@ -88,12 +88,14 @@ def request(
 
         if json is not None and body is not None:
             raise TypeError(
-                "request got values for both 'json' and 'body', can only specify one. Prefer json"
+                "request got values for both 'json' and 'body', can only specify one."
             )
 
         if json is not None:
             if headers is None:
-                headers = self.headers.setdefault("Content-Type", "application/json")  # type: ignore
+                headers = self.headers
```

Comment:
I think this needs to be `self.headers.copy()` or similar, otherwise we'll be mutating the self's default headers in the next line that will persist to future requests.